### PR TITLE
Extract axioms from spec module and add to definition

### LINKED ISSUE #2930: Allow simplification rules in spec file
See https://github.com/kframework/k/issues/2287#issuecomment-965113610

### PR DESCRIPTION
**Pull request by ana-pantilie**
Fixes #2930

### Scope:
- extracts axioms from the spec.kore module and adds them to the main definition
- tested with https://github.com/kframework/k/pull/2345

### Estimate:
- 9 Dec 2021

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/app/share/GlobalMain.hs [hunk 1]
```diff
@@ -560,3 +565,17 @@ mainPatternParseAndVerify indexedModule patternFileName =
...
+-}
+addExtraAxioms ::
+    VerifiedModule StepperAttributes ->
+    VerifiedModule StepperAttributes ->
+    VerifiedModule StepperAttributes
+addExtraAxioms definitionModule moduleWithExtraAxioms =
+    Lens.over
+        (field @"indexedModuleAxioms")
+        (<> indexedModuleAxioms moduleWithExtraAxioms)
+        definitionModule
```
Comment (dopamane) [inline kore/app/share/GlobalMain.hs:581]: Alternatives:

Without generic-lens:
```haskell
let definitionAxioms = indexedModuleAxioms definitionModule
    extraAxioms = indexedModuleAxiom moduleWithExtraAxioms
 in definitionModule{ indexedModuleAxioms = definitionAxioms <> extraAxioms }
```

With generic-lens:
Lens provides an operator [`<>~`](https://hackage.haskell.org/package/lens-5.1/docs/Control-Lens-Operators.html#v:-60--62--126-) to target semigroups: 
```haskell
definitionModule & field @"indexedModuleAxioms <>~ indexedModuleAxioms moduleWithExtraAxioms
```

All three are fine by me, I'm just being thorough. I like the last option because it is specific to semigroups. `let .. in` is a bit wordy but it avoids the extra generic-lens dependency.

**Review by dopamane — APPROVED** [review]

File: kore/app/share/GlobalMain.hs (hunk 1 above)
Comment (ana-pantilie) [reply kore/app/share/GlobalMain.hs]: I added your second suggestion.

**Review by andreiburdusa — APPROVED** [review]
